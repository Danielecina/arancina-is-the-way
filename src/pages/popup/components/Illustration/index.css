.illustration-wrapper {
  cursor: pointer;
}

.fryingPan {
  height: auto;
  position: absolute;
  top: 20px;
  width: 430px;
  left: -90px;
  transition: transform 200ms ease-in-out, opacity 200ms ease-in;
  opacity: 0;
}

.aura {
  width: auto;
  height: 150px;
  position: absolute;
  top: 47px;
  left: 63px;
  opacity: 0;
  transition: opacity 200ms ease;
}

.arancina {
  width: 80%;
  left: 10%;
  height: auto;
  position: absolute;
  top: 45px;
  transform: scale(1.2);
  transition: transform 300ms ease;
}

/* Arancina animation on-fire */
.fryingPan.on-fire {
  opacity: 1;
  transform: translateX(10px) rotate(-5deg);
}

.arancina.on-fire {
  transform: scale(1) rotate(-25deg) translateX(-25px);
}

.aura.on-fire {
  opacity: 1
}
/* End Arancina animation on-fire */


/* Arancina animation */
@keyframes arancina-body {
  0% {
    transform: rotate(360deg)
  }
  100% {
    transform: rotate(0deg);
  }
}

/* Arancina SVG styles */
.arancina .arancina-body {
  box-shadow: 0 0 50px -4px #ffecd2;
}

.arancina .arancina-body,
.arancina .crumb {
  animation-name: arancina-body;
  animation-timing-function: linear;
  animation-iteration-count: infinite;
  transform-origin: 87% 47%;
}

.arancina .arancina-body {
  animation-duration: 80s;
}

.arancina .crumb {
  animation-duration: 35s;
}
/* End Arancina animation */
