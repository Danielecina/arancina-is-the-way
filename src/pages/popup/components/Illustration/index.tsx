import React, {useCallback} from 'react'
import PropTypes from 'prop-types'

import {ReactComponent as FryingPan} from '../../../../assets/svgs/fryingPan.svg'
import {ReactComponent as AuraEffect} from '../../../../assets/svgs/aura.svg'
import {ReactComponent as Arancina} from '../../../../assets/svgs/arancina.svg'

import './index.css'

Illustration.propTypes = {
  watchMode: PropTypes.bool
}
export default function Illustration ({watchMode}) {
  const onFire = useCallback((name) => {
    return watchMode ? `${name} on-fire` : name
  }, [watchMode])

  return (
    <div className={'illustration-wrapper'} title={'arancina-illustration'}>
      <FryingPan className={onFire('fryingPan')} />
      <AuraEffect className={onFire('aura')} />
      <Arancina className={onFire('arancina')} />
    </div>
  )
}
