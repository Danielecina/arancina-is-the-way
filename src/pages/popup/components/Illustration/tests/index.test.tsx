import React from 'react'
import renderer from 'react-test-renderer'

import Illustration from '..'

describe('Toolbar', () => {
  test('snapshot', () => {
    const element = renderer.create(<Illustration />).toJSON()
    expect(element).toMatchSnapshot()
  })

  test('snapshot with watchMode true', () => {
    const element = renderer.create(<Illustration watchMode />).toJSON()
    expect(element).toMatchSnapshot()
  })
})
