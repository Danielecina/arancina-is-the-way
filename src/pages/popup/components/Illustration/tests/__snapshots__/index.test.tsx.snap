// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar snapshot 1`] = `
<div
  className="illustration-wrapper"
>
  <svg
    className="fryingPan"
  >
    fryingPan.svg
  </svg>
  <svg
    className="aura"
  >
    aura.svg
  </svg>
  <svg
    className="arancina"
  >
    arancina.svg
  </svg>
</div>
`;

exports[`Toolbar snapshot with watchMode true 1`] = `
<div
  className="illustration-wrapper"
>
  <svg
    className="fryingPan on-fire"
  >
    fryingPan.svg
  </svg>
  <svg
    className="aura on-fire"
  >
    aura.svg
  </svg>
  <svg
    className="arancina on-fire"
  >
    arancina.svg
  </svg>
</div>
`;
